subsys/bluetooth/host/settings.c

```diff
@@ -240,6 +240,17 @@ static int commit(void)
 		bt_set_name(CONFIG_BT_DEVICE_NAME);
 	}
 #endif
+
+	if (IS_ENABLED(CONFIG_BT_DEVICE_APPEARANCE_DYNAMIC)) {
+		if (!bt_dev.id_count) {
```

Comment:
Why do we check for the `bt_dev.id_count`? Can a device have an appearance per ID?

Reply:
We want to detect that appearance is not set and initialize it. AFAIK 0x0000 is an appearance value that may be intentionally set by the application. I assume `bt_dev.id_count == 0` is true only at first boot, at which time the application can not yet have written the appearance, so it's safe to initialize. I may be misunderstanding something here.

We could have a extra bit that signifies initialization. Or do/should we have an API for initializing settings? Maybe just wrap `bt_dev.id_count == 0` in a function meaningful function?

Reply:
> We want to detect that appearance is not set and initialize it.

Won't settings be read before the application will get a chance to set the values? 

> AFAIK 0x0000 is an appearance value that may be intentionally set by the application.
Isn't the empty string `\0` also a valid name set by the application? If so, then this check needs to also guard the `bt_set_name` above. 

> We could have a extra bit that signifies initialization. Or do/should we have an API for initializing settings? Maybe just wrap bt_dev.id_count == 0 in a function meaningful function?

Either a new bit, a wrapper (function or macro) or at the very least a comment :)